packages/deployer/src/commands.ts

```diff
@@ -0,0 +1,132 @@
+import ora from 'ora';
+import {
+  parseFiles,
+  parseReceipt,
+  generateServerlessSecretsFile,
+  removeServerlessSecretsFile,
+  checkConfigParameters,
+} from './config';
+import * as ssm from './infrastructure/ssm';
+import { applyTerraformWorkaround } from './infrastructure/terraform';
+import { checkProviderRecords } from './evm/evm';
+import { deployAirnode, removeAirnode } from './infrastructure/serverless';
+import {
+  writeJSONFile,
+  deriveXpub,
+  generateMnemonic,
+  deriveProviderId,
+  shortenProviderId,
+  waitForEnter,
+  deriveMasterWalletAddress,
+} from './util';
+
+export async function deployFirstTime(configPath, securityPath, nodeVersion) {
+  const configParams = await parseFiles(configPath, securityPath);
+  checkConfigParameters(configParams, nodeVersion, 'deploy-first-time');
+  await applyTerraformWorkaround(configParams.region);
+
+  const mnemonic = generateMnemonic();
+  const providerId = deriveProviderId(mnemonic);
+  const providerIdShort = shortenProviderId(providerId);
+  if (await ssm.checkIfProviderIdShortExists(providerIdShort)) {
+    throw new Error('Randomly generated providerIdShort is already used at SSM. This should not have happened.');
+  }
+
+  ora().warn('Write down the 12 word-mnemonic below on a piece of paper and keep it in a safe place\n');
```

Comment:
What I don't like the most about all this is how tightly the business logic is coupled with the user interaction bits.

Reply:
Not that I know of